jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/AbstractLatencyRecordingHandler.java

```diff
@@ -0,0 +1,86 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import java.util.function.Function;
+
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.HttpStream;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.NanoTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A <code>Handler</code> that allows recording the latencies of the requests executed by the wrapped handler.
```

Comment:
I think the javadoc needs to be more specific about which latency is being measured.  This is the latency from the first notice of the request (obtained from the stream timestamp)  until after all processing has completed and all serialized methods have returned and the stream completion even has been handled (stream completion).

This is different from the handling latency, which is from when the handler is called until the passed callback is completed.

Reply:
I added some javadoc more precisely describing what we mean with 'latency' in this context. Hopefully it's clear enough.